access/app/validators/institutional_email_validator.rb

```diff
@@ -0,0 +1,19 @@
+class InstitutionalEmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless is_institutional(value)
+      record.errors[attribute] << "is not an institutional mail"
+    end
+  end
+
+  private
+  def is_institutional(email)
+    orgs = Organization.all.map {|d| d.domain}
+    orgs << %w(cern.ch upnet.gr)
```

Comment:
Because they are already hardcoded in the previous version of access. We could make this configurable.

> On 07 Dec 2015, at 13:55, Nicolas Liampotis <notifications@github.com> wrote:
> 
> In access/app/validators/institutional_email_validator.rb <https://github.com/grnet/access-ca-portal/pull/96#discussion_r46817381>:
> 
> > @@ -0,0 +1,19 @@
> > +class InstitutionalEmailValidator < ActiveModel::EachValidator
> > +  def validate_each(record, attribute, value)
> > +    unless is_institutional(value)
> > +      record.errors[attribute] << "is not an institutional mail"
> > +    end
> > +  end
> > +
> > +  private
> > +  def is_institutional(email)
> > +    orgs = Organization.all.map {|d| d.domain}
> > +    orgs << %w(cern.ch upnet.gr)
> Why do we hardcode institutional domains:
> 
> orgs << %w(cern.ch upnet.gr)
> —
> Reply to this email directly or view it on GitHub <https://github.com/grnet/access-ca-portal/pull/96/files#r46817381>.
>